improver_tests/calibration/reliability_calibration/test_ApplyReliabilityCalibration.py

```diff
@@ -469,6 +508,113 @@ def test_calibrating_without_single_value_bins(self, warning_list=None):
+                reliability_cube_spatial = threshold_cube.copy()
+                reliability_cube_spatial.add_aux_coord(forecast_point.coord(y_name))
+                reliability_cube_spatial.add_aux_coord(forecast_point.coord(x_name))
+                reliability_cube_list.append(reliability_cube_spatial)
+
+        expected_0 = np.array(
+            [[0.25, 0.3125, 0.375], [0.4375, 0.5, 0.5625], [0.625, 0.6875, 0.75]]
+        )
+        expected_1 = np.array([[0.25, 0.3, 0.35], [0.4, 0.45, 0.5], [0.55, 0.6, 0.65]])
+
+        result = self.plugin.process(
+            test_forecast, reliability_cube_list, point_by_point=True,
+        )
+
+        # check that data is as expected
+        assert_allclose(result[0].data, expected_0)
+        assert_allclose(result[1].data, expected_1)
+
+        # check that coordinates match
+        coords_table = [c.name() for c in test_forecast.coords()]
+        coords_result = [c.name() for c in result.coords()]
+        assert coords_table == coords_result
+
+        self.assertFalse(warning_list)
+
+    def test_calibrating_spot(self, warning_list=None):
+        """Test application of reliability table to spot forecasts."""
+
+        expected_0 = [0.25, 0.4375, 0.625]
+        expected_1 = [0.25, 0.4, 0.55]
+
+        result = self.plugin.process(
+            self.forecast_spot_cube, self.reliability_cube, point_by_point=False,
+        )
+
+        # check that data is as expected
+        assert_allclose(result[0].data, expected_0)
+        assert_allclose(result[1].data, expected_1)
+
+        # check that coordinates match
+        coords_table = [c.name() for c in self.forecast_spot_cube.coords()]
+        coords_result = [c.name() for c in result.coords()]
+        assert coords_table == coords_result
+
+        self.assertFalse(warning_list)
+
+    def test_calibrating_spot_point_by_point(self, warning_list=None):
+        """Test application of reliability table to spot forecasts using
+        point_by_point functionality."""
+
+        y_name = self.forecast_spot_cube.coord(axis="y").name()
+        x_name = self.forecast_spot_cube.coord(axis="x").name()
+
+        # create reliability table in same format as that output from
+        # ManipulateReliabilityTable when point_by_point=True
+        reliability_table = self.reliability_cubelist.copy()
+        reliability_cube_list = iris.cube.CubeList()
+        for threshold_cube in reliability_table:
+            threshold_cube.remove_coord(y_name)
+            threshold_cube.remove_coord(x_name)
+            for forecast_point in self.forecast_spot_cube.slices_over([y_name, x_name]):
+                reliability_cube_spatial = threshold_cube.copy()
+                reliability_cube_spatial.add_aux_coord(forecast_point.coord(y_name))
+                reliability_cube_spatial.add_aux_coord(forecast_point.coord(x_name))
+                reliability_cube_list.append(reliability_cube_spatial)
```

Comment:
As this code is a duplication of the code in `calibrating_point_by_point` you could consider putting this into a small function at the top of this file to reduce duplication.

Reply:
I've added a function `create_point_by_point_reliability_table` to reduce this duplication